### PR TITLE
Validate inputs only for repo action, no warning for small delay.

### PR DESCRIPTION
**Pull request by TingluoHuang**
https://github.com/github/c2c-actions-service/issues/1051

### CODE REVIEW

File: src/Runner.Worker/ActionRunner.cs [hunk 1]
```diff
@@ -170,11 +170,15 @@ public async Task RunAsync()
                 }
             }
 
-            foreach (var input in userInputs)
+            // Validate inputs only for actions with action.yml
+            if (Action.Reference.Type == Pipelines.ActionSourceType.Repository)
```
Comment (TingluoHuang) [inline src/Runner.Worker/ActionRunner.cs:174]: The real diff is:
```
if (Action.Reference.Type == Pipelines.ActionSourceType.Repository)
{
   existing code
}
```

Comment (TingluoHuang) [reply src/Runner.Worker/ActionRunner.cs:174]: the current cause unwanted warning for `run` step.
ex: https://github.com/actions/canary/actions/runs/102763474
image

File: src/Runner.Worker/ExecutionContext.cs [hunk 2]
```diff
@@ -335,7 +336,7 @@ public TaskResult Complete(TaskResult? result = null, string currentOperation =
             }
 
             // report total delay caused by server throttling.
-            if (_totalThrottlingDelayInMilliseconds > 0)
+            if (_totalThrottlingDelayInMilliseconds > _throttlingDelayReportThreshold)
```
Comment (ericsciple) [inline src/Runner.Worker/ExecutionContext.cs:339]: Discussed offline, this is related to log lines and timeline record updates. Not general perf tarpitting.

Upload artifact, etc would be reported as 429 during that experience (depends on whether they log).

Comment (ericsciple) [reply src/Runner.Worker/ExecutionContext.cs:339]: Existing code so better to just leave it, consider Timespan in future so name doesn't need to convey units :)

File: src/Runner.Worker/ActionRunner.cs (hunk 1 above)
Comment (ericsciple) [reply src/Runner.Worker/ActionRunner.cs:174]: cool, basically anyhting with action.yml

**Comment by TingluoHuang** [comment]
https://github.com/actions/runner/issues/477

**Pull request merged**